apps/manager_app/auth_card.c

```diff
@@ -0,0 +1,542 @@
+    manager_auth_card_response_t *resp);
+
+/**
+ * @brief Helper to return signature of server challenge.
+ * Handles challenge signing task and populates resp object with appropriate
+ * response
+ *
+ * @param challenge_request object pointer with challenge query for card auth
+ * @param resp object pointer card auth response
+ * @return manager_error_code_t MANAGER_TASK_SUCCESS if successful, else error
+ * code
+ */
+manager_error_code_t handle_sign_challenge(
+    manager_auth_card_challenge_request_t *challenge_request,
+    manager_auth_card_response_t *resp);
+
+/**
+ * @brief Helper to handle initiate query for auth card.
+ * Hanlder performs following actions
+ * - Get user's confirmation for proceeding with card auth
+ * - Prepare context for auth card flow
+ * - returns sign serial response
+ *
+ * @param initiate_request object pointer with initiate query for card auth
+ * @param resp object pointer card auth response
+ * @return manager_error_code_t MANAGER_TASK_SUCCESS if successful, else error
+ * code
+ */
+manager_error_code_t handle_auth_card_initiate_query(
+    manager_auth_card_initiate_request_t *initiate_request,
+    manager_auth_card_response_t *resp);
+
+/**
+ * @brief Helper to handle challenge query for auth card.
+ * Handles challenge signing task and populates resp object with appropriate
+ * response
+ *
+ * @param challenge_request object pointer with challenge query for card auth
+ * @param resp object pointer card auth response
+ * @return manager_error_code_t MANAGER_TASK_SUCCESS if successful, else error
+ * code
+ */
+manager_error_code_t handle_auth_card_challenge_query(
+    manager_auth_card_challenge_request_t *challenge_request,
+    manager_auth_card_response_t *resp);
+
+/**
+ * @brief Helper to handle result of auth card.
+ * Handles pairing of card based on result, flow state and card pair requirement
+ *
+ * @param result_query object pointer with result for card auth
+ * @return manager_error_code_t MANAGER_TASK_SUCCESS if successful, else error
+ * code
+ */
+manager_error_code_t handle_auth_card_result_query(
+    manager_auth_card_result_t *result_query);
+
+/**
+ * @brief Helper to parse handle queries for auth card.
+ * Hanlder performs following actions
+ * - Decode host query for manager app
+ * - Verify query is for card auth and type of query
+ * - Call relevant handler for queries
+ *
+ * @param usb_evt object to usb_evt with host query
+ * @param resp object pointer card auth response
+ * @return manager_error_code_t MANAGER_TASK_SUCCESS if successful, else error
+ * code
+ */
+manager_error_code_t handle_auth_card_query(usb_event_t *usb_evt,
+                                            manager_auth_card_response_t *resp);
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+auth_card_screen_ctx_t auth_card_ctx = {0};
```

Comment:
Is there any way in which we can make this a local variable and pass it through pointers in the different handlers?

It would help in reducing overall global variables used in the firmware

Reply:
Will updated